### ISSUE TITLE
CompressedTensorsConfig(run_compressed=False) silently random-initializes unmapped fused-MoE expert weights on 5.10.x (works on 5.13.1) — request hard error for unmapped packed tensors

### ISSUE THREAD

**Issue by narfian** [huggingface/transformers#47407]
**Summary**: Loading a compressed-tensors W4A16 checkpoint of `Qwen/Qwen3.6-35B-A3B` (fused-expert MoE) with `from_pretrained(ckpt, quantization_config=CompressedTensorsConfig(run_compressed=False))` on **transformers 5.10.1** prints:

```
model.language_model.layers.{0...39}.mlp.experts.gate_up_proj_packed | UNEXPECTED
model.language_model.layers.{0...39}.mlp.experts.gate_up_proj        | MISSING
model.language_model.layers.{0...39}.mlp.experts.down_proj           | MISSING
```

i.e. the decompressor does not map the packed fused expert tensors, and those weights are **silently newly-initialized** — the returned "trainable QDQ model" produces garbage logits for every MoE layer while `from_pretrained` "succeeds". This breaks QAT/recovery-training workflows. (Diagnostic: the broken student's logits agree 97% with an independently-loaded copy of itself, but 0% with the uncorrupted teacher.)

**Environment (broken)**: transformers 5.10.1, compressed-tensors 0.17.1.

**Resolution verified**: on transformers **5.13.1** the same flow decompresses correctly (per-expert checkpoint tensors are mapped into the fused module via the conversion-mapping loader; we verified the QDQ values match a manual dequant of the packed weights to 2.4e-4 max abs error).

**Request**: on versions/paths where packed tensors cannot be mapped, please **raise a hard error instead of silently newly-initializing weights**. Silent random init behind a successful `from_pretrained` is the worst failure mode for downstream training pipelines.

**Comment by webtecnica** [comment]
Opened **Part 1** of this: #47411

Extends the approach from #47408 to also cover quantized checkpoints. After `_finalize_model_loading`, if there's a near-total weight mismatch (>99% missing) WITH either `weight_mapping` or `hf_quantizer` registered, a `ValueError` is raised instead of silently random-initializing.

**Part 2** (the actual fused-MoE decompression fix) still needs to be done — tracking the root cause in the `WeightConverter` / `DecompressExperts` pipeline for fused packed tensors like `gate_up_proj_packed`.

cc @CyrilVallez @zucchini-nlp

**Comment by zucchini-nlp** [comment]
Hey @narfian , so iiuc the issue exists on older versions and is resolved now

>  on transformers 5.13.1 the same flow decompresses correctly, on versions/paths where packed tensors cannot be mapped, please raise a hard error 

Unfortunately we cannot change an already released pypi package, and imo a logged error that weights are missing is enought to give clues that smth went wrong in loading. LMK if there is anything else you need help with
